### ISSUE TITLE
Missing window controls

### ISSUE THREAD

**Issue by maymage** [flathub/xyz.rescribe.rescribe#5]
This is how `Rescribe` looks like on my Fedora 38, GNOME 44 machine

![image](https://github.com/flathub/xyz.rescribe.rescribe/assets/36572014/d7e52734-7696-4e24-b9d0-c3c0037b8b17)

This is with the current version

```
flatpak info xyz.rescribe.rescribe 

Rescribe - High quality OCR for images, PDFs and Google Books.

         Kennung: xyz.rescribe.rescribe
             Ref: app/xyz.rescribe.rescribe/x86_64/stable
     Architektur: x86_64
           Zweig: stable
         Version: 1.1.0
          Lizenz: GPL-3.0
        Ursprung: flathub
        Sammlung: org.flathub.Stable
    Installation: system
     Installiert: 194,6 MB
Laufzeitumgebung: org.freedesktop.Platform/x86_64/22.08
             Sdk: org.freedesktop.Sdk/x86_64/22.08

          Commit: 583d051d751e3a8e332528a299fcc6186762e4fc99121700…
          Parent: be08c6d75d763928eb4b494b95517b51e8511bf891370f7b…
         Betreff: Build wayland and x11 versions and use launcher script to choose the appropriate one (567859ef)
           Datum: 2023-02-14 14:12:14 +0000

```

**Comment by nickjwhite** [comment]
Hi @maymage, I'm so sorry I left this issue for so long!

It has been a very long time since I've used Gnome (I use Sway myself), so I missed that the window decorations were missing - thanks so much for your report. I looked into window decoration support in the GUI toolkit we use (fyne), and it looks like the next version (2.5.0) will properly support decorations, [thanks to this commit](https://github.com/fyne-io/fyne/pull/4481). I'll make a new flatpak release of rescribe updating to this once it's out and stable, and will keep this ticket open until then.

Thanks again for reporting this!

**Comment by Jacalz** [comment]
I'd just want to chime in on this. Wayland support is, as of Fyne v2.4.x, considered not fully supported and I would strongly advice against using the Wayland support as it currently stands. The users will get a lot better support with the X11 backend. However, this will hopefully not be the case with v2.5.0 once that releases. The PR mentioned above does improve things a lot but there are still a few minor hiccups left to solve (mostly scaling).

**Comment by nickjwhite** [comment]
Hi @Jacalz, thanks for chiming in. I use wayland myself, and my experience with the Fyne wayland support is that it's perfectly good for Rescribe (with the exception of this issue), and the scaling is perfect, unlike X11 on Wayland. So I'll stick with Wayland unless I get reports of stuff not working for people. The flatpak package does actually include X11 and Wayland builds, and uses a little launcher script to decide which to execute based on what environment it's in (see [line 27](https://github.com/flathub/xyz.rescribe.rescribe/blob/6d75b9af54c422a863db1c90654f760625dae8bd/xyz.rescribe.rescribe.yml#L27)).

**Comment by Jacalz** [comment]
Alright. I'm glad that you find it to be working well. Nice idea with the script to launch the right binary.
I might want to do something similar for https://github.com/Jacalz/rymdport/issues/122.

I'll talk to the rest of the Fyne team if we should consider backporting https://github.com/fyne-io/fyne/pull/4481/files and https://github.com/fyne-io/fyne/pull/4704 for the new versions of GLFW with better Wayland support. I can't guarantee anything but I'll see what they have to say.

**Comment by Jacalz** [comment]
Another thing worth noting: I'm glad to see another Fyne application on Flatpak and Flathub. I regret not knowing about this one when doing the first talk in https://m.youtube.com/watch?v=EsX8CgC_N5M&pp=ygUNZnluZWNvbmYgMjAyMw%3D%3D.

Feel free to reach out if you find any problems with Flatpak, Wayland support or any other issues. Things should be a lot nicer with https://github.com/fyne-io/fyne/pull/4406 in v2.5.0 anyway.

**Comment by nickjwhite** [comment]
Yes absolutely @Jacalz, I'm really looking forward to the v2.5.0 release for the XDG Desktop Portal support and the window controlls stuff working, thanks for working on it!

**Comment by rugk** [comment]
Similar issue in https://github.com/flathub/xyz.rescribe.rescribe/issues/10

**Comment by nickjwhite** [comment]
OK I made a test build, can you try installing it @maymage  and let me know if it fixes this issue? Instructions to install are here: https://github.com/flathub/xyz.rescribe.rescribe/pull/11#issuecomment-3601800159

**Comment by nickjwhite** [comment]
I have just released the new version, which should fix this, so I'll close the bug now. Feel free to re-open it if it's still an issue for you, and thanks again for the report.
